neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/LazyStoreList.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2013-2018 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.core.internal;
+
+import fr.inria.atlanmod.neoemf.core.PersistentEObject;
+import fr.inria.atlanmod.neoemf.data.store.Storable;
+import fr.inria.atlanmod.neoemf.data.store.Store;
+import fr.inria.atlanmod.neoemf.data.store.adapter.StoreAdapter;
+
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.InternalEObject;
+import org.eclipse.emf.ecore.impl.EStoreEObjectImpl;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import static java.util.Objects.isNull;
+
+/**
+ * A {@link List} representing a multi-valued feature which behaves as a proxy and that delegates its operations to the
+ * associated {@link Store}.
+ *
+ * @param <E> the type of elements in this list
+ *
+ * @see PersistentEObject#eStore()
+ */
+@ParametersAreNonnullByDefault
+public class LazyStoreList<E> extends EStoreEObjectImpl.BasicEStoreEList<E> {
+
+    @SuppressWarnings("JavaDoc")
+    private static final long serialVersionUID = 2630358403343923944L;
+
+    /**
+     * The object that holds the store.
+     */
+    @Nonnull
+    private final transient Storable storable;
+
+    /**
+     * Constructs a new {@code LazyStoreList}.
+     *
+     * @param owner   the owner the {@code feature}
+     * @param feature the feature associated with this list
+     */
+    public LazyStoreList(PersistentEObject owner, EStructuralFeature feature) {
+        super(owner, feature);
+        this.storable = owner;
+    }
+
+    @Nonnull
+    @Override
+    protected StoreAdapter eStore() {
+        return storable.eStore();
+    }
+
+    // region Delegating methods
+
+    @Override
+    protected void delegateAdd(Object object) {
+        delegateAdd(InternalEObject.EStore.NO_INDEX, object);
+    }
+
+    @Override
+    protected boolean delegateContainsAll(Collection<?> collection) {
+        if (isNull(collection) || collection.isEmpty()) {
+            return false;
+        }
+
+        if (collection.size() <= 1) {
+            return contains(collection.iterator().next());
+        }
+
+        return eStore().getAll(owner, eStructuralFeature).containsAll(collection);
+    }
+
+    @Override
+    // FIXME Does not resolve
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)